### ISSUE TITLE
Recurrent dropout

### ISSUE THREAD

**Issue by varshaneya** [pytorch/pytorch#15987]
## 🚀 Feature
Recurrent Dropout for RNN, GRU and LSTM.

## Motivation
The concept of dropout to sequence models has been proposed in this [paper](https://arxiv.org/pdf/1512.05287.pdf). Also popular libraries like Keras and Tensorflow have their native implemenation of this but Pytorch does not. I personally feel Pytorch is giving a very stiff competition to Keras and Tensorflow, hence must not leave out this feature.

## Pitch
There has been some good results in the [paper](https://arxiv.org/pdf/1512.05287.pdf) for using droupouts in sequence models for language modelling. Hence adding this to Pytorch would be of great use to Pytorch users rather than depending on alternative solution.

## Alternatives
An alternate solution is to depend on a non-native implementation of the same is given [here](https://github.com/salesforce/awd-lstm-lm/) as part of the language modelling toolkit of Salesforce.

**Comment by PetrochukM** [comment]
Similar to https://github.com/pytorch/pytorch/issues/9572

**Comment by zou3519** [comment]
Closing this because it is a duplicate, please voice your support in #9572
